### PR TITLE
PSI: rework for future version

### PR DESCRIPTION
**Pull request by wpoely86**
They switched to a CMake based build system

The plugin stuff doesn't yet work but that will probably need patching in the easyconfig.

### CODE REVIEW

**Comment by boegel** [comment]
@wpoely86: I'd like to see a test report of reinstalling the existing PSI easyconfigs on top of this modified easyblock...

File: easybuild/easyblocks/p/psi.py [hunk 1]
```diff
@@ -138,7 +156,7 @@ def sanity_check_step(self):
         """Custom sanity check for PSI."""
         custom_paths = {
             'files': ['bin/psi%s' % self.version.split('.')[0]],
-            'dirs': ['include', 'share/psi'],
+            'dirs': ['share/psi'],
```
Comment (boegel) [inline easybuild/easyblocks/p/psi.py]: why this change?

File: easybuild/easyblocks/p/psi.py [hunk 2]
```diff
@@ -121,7 +87,59 @@ def configure_step(self):
...
 
-        super(EB_PSI, self).configure_step(cmd_prefix=self.cfg['start_dir'])
+        # explicitely specify Python binary to use
+        pythonroot = get_software_root('Python')
+        if not pythonroot:
+            self.log.error("Python module not loaded.")
+        env.setvar('PYTHON', os.path.join(pythonroot, 'bin', 'python'))
+
+        # They have a non-standard variable to define the install path
+        env.setvar('PREFIX', self.installdir)
```
Comment (boegel) [inline easybuild/easyblocks/p/psi.py]: this wasn't there before, so specific to the recent versions using `CMake`?

**Comment by boegel** [comment]
Looks pretty good overall.

What do you mean with your comment w.r.t. the plugin support exactly? Which easyconfigs will need patching? Or are we talking a patch file that will be required in order to enable plugin support?

File: easybuild/easyblocks/p/psi.py (hunk 1 above)
Comment (wpoely86) [inline easybuild/easyblocks/p/psi.py]: They no longer install the headers like this. No idea if that's intentional or a bug in their new build system

File: easybuild/easyblocks/p/psi.py (hunk 2 above)
Comment (wpoely86) [inline easybuild/easyblocks/p/psi.py]: Older version also had an CMake based build system, but it was optional.

I don't see the harm in doing this in all cases?

**Comment by wpoely86** [comment]
@boegel the `psi4 --new-plugin` will generate a makefile that doesn't work. That still needs fixing but there are issues with their build system.

**Comment by boegel** [comment]
Existing PSI easyconfigs are broken because of the changes made here:

```
configure: error: unrecognized option: -DENABLE_MPI=ON
```

**Comment by wpoely86** [comment]
@boegel fixed

**Comment by boegel** [comment]
@wpoely86: still issues, for example with the existing `PSI-4.0b4-ictce-4.1.13-mt.eb`:

```
== 2015-03-01 16:02:39,063 main.EB_PSI WARNING Sanity check: no file of ('lib/libplugin.a',) in ...
...
== 2015-03-01 16:02:39,064 main.EB_PSI WARNING Sanity check: no file of ('include/libplugin/plugin.h',) in ...
...
== 2015-03-01 16:02:39,065 main.EB_PSI WARNING Sanity check: no dir of ('lib',) in ...
```

**Comment by wpoely86** [comment]
@boegel okay, I restored the original sanity check. Should be fine now.

**Comment by boegel** [comment]
Tested with https://github.com/hpcugent/easybuild-easyconfigs/pull/1431 and all existing PSI easyconfigs, works like a charm (now).

Going in, thanks @wpoely86!
